internal/fields/model.go

```diff
@@ -82,3 +89,67 @@ func updateFields(origFields, fields []FieldDefinition) []FieldDefinition {
 	}
 	return updatedFields
 }
+
+// FieldDefinitions is an array of FieldDefinition, this can be unmarshalled from
+// a yaml list or a yaml map.
+type FieldDefinitions []FieldDefinition
+
+func (fds *FieldDefinitions) UnmarshalYAML(value *yaml.Node) error {
+	nilNode := yaml.Kind(0)
+	switch value.Kind {
+	case yaml.SequenceNode:
+		// Fields are defined as a list, this happens in Beats fields files.
+		var fields []FieldDefinition
+		err := value.Decode(&fields)
+		if err != nil {
+			return err
+		}
+		*fds = fields
+		return nil
+	case yaml.MappingNode:
+		// Fields are defined as a map, this happens in ecs fields files.
+		var fields []FieldDefinition
+		for i := 0; i < len(value.Content); i += 2 {
```

Comment:
Is it safe to assume it's always a pair? Is there a chance that it will become a triple :)? a comment, or second value...

Reply:
In principle yes, the content of a mapping node should be pairs of key/values. Comments are included in the significative nodes themselves.

But let me add an additional constraint here to avoid reading the `i+1` value when it doesn't exist.